casper/src/test/scala/coop/rchain/casper/util/rholang/MergingBranchMergerSpec.scala

```diff
@@ -339,64 +335,77 @@ class MergingBranchMergerSpec extends FlatSpec with Matchers {
+
+            s = s"merging ${mergingBlocks.size} children into ${PrettyPrinter.buildString(startPreStateHash)}"
+            _ = println(s)
+
+            indices <- (base +: mergingBlocks)
+                        .traverse(
+                          b =>
+                            BlockIndex(
+                              b.blockHash,
+                              b.body.deploys,
+                              Blake2b256Hash.fromByteString(b.body.state.preStateHash),
+                              Blake2b256Hash.fromByteString(b.body.state.postStateHash),
+                              runtimeManager.getHistoryRepo
+                            ).map((b.blockHash -> _))
+                        )
+                        .map(_.toMap)
 
-            deployChainIndex = (v: MergingVertex) =>
-              BlockIndex[Task, Par, BindPattern, ListParWithRandom, TaggedContinuation](
-                v.blockHash,
-                v.processedDeploys.toList,
-                Blake2b256Hash.fromByteString(v.preStateHash),
-                Blake2b256Hash.fromByteString(v.postStateHash),
-                runtimeManager.getHistoryRepo
-              ).map(_.deployChains)
+            _   <- inMemBlockDagStorage.addFinalizedBlockHash(base.blockHash)
+            dag <- inMemBlockDagStorage.getRepresentation
             // merge children to get next preStateHash
-            v <- DagMerger.merge[Task, MergingVertex](
-                  mergingTips,
-                  base,
+            v <- DagMerger.merge[Task](
                   dag,
-                  v => (if (v == base) true else false).pure,
-                  v => deployChainIndex(v),
-                  v => Blake2b256Hash.fromByteString(v.postStateHash),
+                  base.blockHash,
+                  Blake2b256Hash.fromByteString(base.body.state.postStateHash),
+                  indices(_).deployChains.pure,
                   runtimeManager.getHistoryRepo
                 )
-            (nextPreStateHash, rejectedDeploys) = v
-            _                                   = assert(rejectedDeploys.size == 0)
-            _                                   = println(s"merge result ${PrettyPrinter.buildString(nextPreStateHash)}")
+            (postState, rejectedDeploys) = v
+            nextPreStateHash             = ByteString.copyFrom(postState.bytes.toArray)
+            _                            = assert(rejectedDeploys.size == 0)
+            _                            = assert(nextPreStateHash != base.body.state.postStateHash)
+            _                            = println(s"merge result ${PrettyPrinter.buildString(nextPreStateHash)}")
           } yield nextPreStateHash
 
         for {
-          // create base state - child of genesis
-          _ <- List.range(0, 4).foldLeftM(genesisPostStateHash) { (baseHash, i) =>
-                mkLayer(baseHash, i)
-              }
+          dagStore <- InMemBlockDagStorage.create[Task]
```

Comment:
You can fix this test with with in-memory BlockDagKeyValueStorage.
https://github.com/rchain/rchain/blob/2d5aac8a270ee8053240eb340db9bb3624b52372/casper/src/test/scala/coop/rchain/casper/engine/Setup.scala#L119-L121

For easier sync with dev just undo changes to InMemBlockDagStorage and we can merge the PR. :tada: